hooks/useAsync.ts

```
@@ -0,0 +1,41 @@
+      const result = await fn(controller.signal);
+      setState({ status: 'success', data: result });
+      return result;
+    } catch (e) {
+      const appErr = toAppError(e);
+      setState({ status: 'error', error: appErr });
+      return undefined;
+    } finally {
+      abortRef.current = null;
+      if (options?.autoResetMs) {
+        const timeout = setTimeout(() => setState({ status: 'idle' }), options.autoResetMs);
+        return () => clearTimeout(timeout);
```

Biome (2.1.2)

[error] Unsafe usage of 'return'.

'return' in 'finally' overwrites the control flow statements inside 'try' and 'catch'.

(lint/correctness/noUnsafeFinally)